### PR TITLE
bear: 3.1.3 -> 3.1.5

### PR DESCRIPTION
**Pull request by xokdvium**
Release notes: https://github.com/rizsotto/Bear/releases/tag/3.1.5

Cleaned the code related to tests:
- Split tests into `checkPhase` by applying a tiny patch to `CMakeLists.txt`.
- Bumped protobuf, which was blocked by a build failure fixed in https://github.com/rizsotto/Bear/commit/3ba056a6cdebd1608279e89b021e7c28bcee3ea3
- Moved to pkgs/by-name

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by xokdvium** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>bear</li>
  </ul>
</details>

**Review by DieracDelta — APPROVED** [review]
Thanks for this! Can confirm bear builds on aarch64-darwin. LGTM

File: pkgs/development/tools/build-managers/bear/fix-functional-tests.patch
[the remark this answers is not in the document]
Comment (xokdvium) [reply pkgs/development/tools/build-managers/bear/fix-functional-tests.patch:1]: This patch really did two things:

1. Hardcoded `True` for is_preload_disabled for darwin, which upstream has already done in https://github.com/rizsotto/Bear/commit/8bf9f76cb67497cfe745e2065d3e419e7edbc08c
2. Passing NIX_* related variables into the lit environment, but I doesn't seem to be doing anything. The tests I had to disable are failing either way regardless of this change.

File: pkgs/by-name/be/bear/package.nix [hunk 1]
```diff
@@ -20,54 +20,81 @@
...
   pname = "bear";
-  version = "3.1.3";
+  version = "3.1.5";
 
   src = fetchFromGitHub {
     owner = "rizsotto";
-    repo = pname;
-    rev = version;
-    hash = "sha256-1nZPzgLWcmaRkOUXdm16IW2Nw/p1w8GBGEfZX/v+En0=";
+    repo = "bear";
```
[the remark this answers is not in the document]
Comment (xokdvium) [reply pkgs/by-name/be/bear/package.nix:29]: Using `finalAttrs.pname` is discouraged: https://github.com/NixOS/nixpkgs/issues/277994

**Pull request merged**